lightning-background-processor/src/lib.rs

```diff
@@ -101,25 +104,31 @@ impl BackgroundProcessor {
 	/// [`FilesystemPersister::persist_manager`]: lightning_persister::FilesystemPersister::persist_manager
 	pub fn start<
 		Signer: 'static + Sign,
-		M: 'static + Deref + Send + Sync,
+		CF: 'static + Deref + Send + Sync,
+		CW: 'static + Deref + Send + Sync,
 		T: 'static + Deref + Send + Sync,
 		K: 'static + Deref + Send + Sync,
 		F: 'static + Deref + Send + Sync,
 		L: 'static + Deref + Send + Sync,
+		P: 'static + Deref + Send + Sync,
 		Descriptor: 'static + SocketDescriptor + Send + Sync,
 		CMH: 'static + Deref + Send + Sync,
 		RMH: 'static + Deref + Send + Sync,
-		CMP: 'static + Send + ChannelManagerPersister<Signer, M, T, K, F, L>,
-		CM: 'static + Deref<Target = ChannelManager<Signer, M, T, K, F, L>> + Send + Sync,
+		EH: 'static + EventHandler + Send + Sync,
+		CMP: 'static + Send + ChannelManagerPersister<Signer, CW, T, K, F, L>,
+		M: 'static + Deref<Target = ChainMonitor<Signer, CF, T, F, L, P>> + Send + Sync,
+		CM: 'static + Deref<Target = ChannelManager<Signer, CW, T, K, F, L>> + Send + Sync,
 		PM: 'static + Deref<Target = PeerManager<Descriptor, CMH, RMH, L>> + Send + Sync,
 	>
-	(handler: CMP, channel_manager: CM, peer_manager: PM, logger: L) -> Self
+	(persister: CMP, event_handler: EH, chain_monitor: M, channel_manager: CM, peer_manager: PM, logger: L) -> Self
 	where
-		M::Target: 'static + chain::Watch<Signer>,
+		CF::Target: 'static + chain::Filter,
+		CW::Target: 'static + chain::Watch<Signer>,
 		T::Target: 'static + BroadcasterInterface,
 		K::Target: 'static + KeysInterface<Signer = Signer>,
 		F::Target: 'static + FeeEstimator,
 		L::Target: 'static + Logger,
+		P::Target: 'static + channelmonitor::Persist<Signer>,
 		CMH::Target: 'static + ChannelMessageHandler,
 		RMH::Target: 'static + RoutingMessageHandler,
```

Comment:
Lol a casual 30-line method signature... Curious what the "best practice" for this would be. Definitely has been an issue since before this PR though.

Reply:
Not sure honestly unless we want to use trait objects. :(